soc/nordic/nrf54h/soc.c

```diff
@@ -153,7 +153,9 @@ void soc_early_init_hook(void)
 	nrf_spu_periph_perm_dmasec_set(spu, nrf_address_slave_get(ccm030_addr), true);
 #endif
 
-	if (DT_NODE_HAS_STATUS(DT_NODELABEL(nfct), disabled) &&
+	if (((IS_ENABLED(CONFIG_SOC_NRF54H20_CPUAPP) &&
+	      DT_NODE_HAS_STATUS(DT_NODELABEL(nfct), disabled)) ||
+	     DT_NODE_HAS_STATUS(DT_NODELABEL(nfct), reserved)) &&
```

Comment:
I think it should be mentioned somewhere (maybe in the `nordic,nrf-nfct-v2` binding, in the `nfct-pins-as-gpios` property description) that "reserved" should be used as the node status, so that the NFCT peripheral is assigned to a given core.